User_Service/main.go

```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	controllers "github.com/E-Furqan/Food-Delivery-System/Controllers"
+	config "github.com/E-Furqan/Food-Delivery-System/database_config"
+	"github.com/E-Furqan/Food-Delivery-System/middleware"
+	"github.com/gin-gonic/gin"
+)
+
+func main() {
+	config.Connection()
+	server := gin.Default()
+
+	server.POST("/Register", controllers.Register)
+	server.GET("/Getuser", controllers.Get_user)
+	server.GET("/Getrole", controllers.Get_role)
+	server.POST("/Login", controllers.Login)
+
+	protected := server.Group("/protected")
+	protected.Use(middleware.AuthMiddleware())
+	{
+		protected.PATCH("/Update_role", controllers.Update_Role)
+		protected.PATCH("/Update_user", controllers.Update_user)
+		protected.DELETE("/Delete_user", controllers.Delete_user)
+		protected.DELETE("/Delete_role", controllers.Delete_role)
+	}
+	server.Run(":8081")
```

Comment:
Why a /protected/ endpoint?

Wouldn't this make more sense:
app/user/update

Reply:
i will update the api endpoint such that they make more sense.

Reply:
have made the changes to  app/Login
app/user/Delete
app/role/Update

Reply:
we are calling user service endpoints, is there some better way to do this?

Reply:
we can create a separate folder for the routes to keep the main file clean.